cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -614,14 +659,30 @@ func (m *gceManagerImpl) GetMigTemplateNode(mig Mig) (*apiv1.Node, error) {
 
 // parseMIGAutoDiscoverySpecs returns any provided NodeGroupAutoDiscoverySpecs
 // parsed into configuration appropriate for MIG autodiscovery.
+// allows either label-based or name-prefix-based entries.
+// returns an error if both label-based and name-prefix-based entries are specified
+// returns an error if more than 1 label-based entry is specified
 func parseMIGAutoDiscoverySpecs(o cloudprovider.NodeGroupDiscoveryOptions) ([]migAutoDiscoveryConfig, error) {
-	cfgs := make([]migAutoDiscoveryConfig, len(o.NodeGroupAutoDiscoverySpecs))
-	var err error
-	for i, spec := range o.NodeGroupAutoDiscoverySpecs {
-		cfgs[i], err = parseMIGAutoDiscoverySpec(spec)
+	var cfgs []migAutoDiscoveryConfig
+	var labelEntryFound bool
+	for _, spec := range o.NodeGroupAutoDiscoverySpecs {
+		c, err := parseMIGAutoDiscoverySpec(spec)
```

Comment:
nit, optional: I think you could simplify the error handling here a bit by first determining the discoverer per spec and then comparing discoverer counts.